packages/@angular/cli/models/webpack-configs/styles.ts

```diff
@@ -44,39 +44,60 @@ export function getStylesConfig(wco: WebpackConfigOptions) {
+      filter: ({ url }: { url: string}) => url.startsWith('/') && !url.startsWith('//'),
+      url: ({ url }: { url: string }) => {
+        if (deployUrl.match(/:\/\//) || deployUrl.startsWith('/')) {
+          // If deployUrl is absolute or root relative, ignore baseHref & use deployUrl as is.
+          return `${deployUrl.replace(/\/$/, '')}${url}`;
+        } else if (baseHref.match(/:\/\//)) {
+          // If baseHref contains a scheme, include it as is.
+          return baseHref.replace(/\/$/, '') +
+              `/${deployUrl}/${url}`.replace(/\/\/+/g, '/');
+        } else {
+          // Join together base-href, deploy-url and the original URL.
+          // Also dedupe multiple slashes into single ones.
+          return `/${baseHref}/${deployUrl}/${url}`.replace(/\/\/+/g, '/');
+        }
+      }
+    };
+
+    config.push(convertRelativeUrlConfig);
+
+    // Do not inline any assets if the inline-asset-max-size option is negative
+    if (!buildOptions.inlineAssetMaxSize || buildOptions.inlineAssetMaxSize >= 0) {
+      const inlineAssetsConfig: PostcssUrlConfig = {
+        // TODO: inline .cur if not supporting IE (use browserslist to check)
+        filter: (asset: any) => !asset.hash && !asset.absolutePath.endsWith('.cur'),
+        url: 'inline',
+        // NOTE: maxSize is in KB
+        maxSize: Number.isInteger(buildOptions.inlineAssetMaxSize) ?
+          buildOptions.inlineAssetMaxSize : 10,
+      };
+
+      config.push(inlineAssetsConfig);
+    }
+
+    return config;
+  };
+
   const postcssPluginCreator = function() {
     return [
       postcssUrl({
         filter: ({ url }: { url: string }) => url.startsWith('~'),
         url: ({ url }: { url: string }) => path.join(projectRoot, 'node_modules', url.substr(1)),
       }),
-      postcssUrl([
-        {
-          // Only convert root relative URLs, which CSS-Loader won't process into require().
-          filter: ({ url }: { url: string }) => url.startsWith('/') && !url.startsWith('//'),
-          url: ({ url }: { url: string }) => {
-            if (deployUrl.match(/:\/\//) || deployUrl.startsWith('/')) {
-              // If deployUrl is absolute or root relative, ignore baseHref & use deployUrl as is.
-              return `${deployUrl.replace(/\/$/, '')}${url}`;
-            } else if (baseHref.match(/:\/\//)) {
-              // If baseHref contains a scheme, include it as is.
-              return baseHref.replace(/\/$/, '') +
-                  `/${deployUrl}/${url}`.replace(/\/\/+/g, '/');
-            } else {
-              // Join together base-href, deploy-url and the original URL.
-              // Also dedupe multiple slashes into single ones.
-              return `/${baseHref}/${deployUrl}/${url}`.replace(/\/\/+/g, '/');
-            }
-          }
-        },
-        {
-          // TODO: inline .cur if not supporting IE (use browserslist to check)
-          filter: (asset: any) => !asset.hash && !asset.absolutePath.endsWith('.cur'),
```

Comment:
Instead of the extra function (which is breaking the eject command).  Add a condition to the filter here.
Also the semantics of the option values should be changed.  A maximum size of zero (0) means nothing should be inlined.  Infinity would mean everything should be inlined.